wrappers/golang/pqc/ml-kem/kyber_params.go

```diff
@@ -0,0 +1,93 @@
+	Dv:                4,
+
+	KeygenFFI: func(cEntropy *C.uint8_t, cConfig *C.MlKemConfig, cPublicKeys *C.uint8_t, cSecretKeys *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_keygen512(cEntropy, cConfig, cPublicKeys, cSecretKeys))
+	},
+	EncapsulateFFI: func(cMessage *C.uint8_t, cPublicKeys *C.uint8_t, cConfig *C.MlKemConfig, cCiphertexts *C.uint8_t, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_encapsulate512(cMessage, cPublicKeys, cConfig, cCiphertexts, cSharedSecrets))
+	},
+	DecapsulateFFI: func(cSecretKeys *C.uint8_t, cCiphertexts *C.uint8_t, cConfig *C.MlKemConfig, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_decapsulate512(cSecretKeys, cCiphertexts, cConfig, cSharedSecrets))
+	},
+}
+
+var Kyber768Params = KyberParams{
+	PublicKeyBytes:    1184,
+	SecretKeyBytes:    2400,
+	CiphertextBytes:   1088,
+	SharedSecretBytes: 32,
+	K:                 3,
+	Eta1:              2,
+	Eta2:              2,
+	Du:                10,
+	Dv:                4,
+
+	KeygenFFI: func(cEntropy *C.uint8_t, cConfig *C.MlKemConfig, cPublicKeys *C.uint8_t, cSecretKeys *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_keygen768(cEntropy, cConfig, cPublicKeys, cSecretKeys))
+	},
+	EncapsulateFFI: func(cMessage *C.uint8_t, cPublicKeys *C.uint8_t, cConfig *C.MlKemConfig, cCiphertexts *C.uint8_t, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_encapsulate768(cMessage, cPublicKeys, cConfig, cCiphertexts, cSharedSecrets))
+	},
+	DecapsulateFFI: func(cSecretKeys *C.uint8_t, cCiphertexts *C.uint8_t, cConfig *C.MlKemConfig, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_decapsulate768(cSecretKeys, cCiphertexts, cConfig, cSharedSecrets))
+	},
+}
+
+var Kyber1024Params = KyberParams{
+	PublicKeyBytes:    1568,
+	SecretKeyBytes:    3168,
+	CiphertextBytes:   1568,
+	SharedSecretBytes: 32,
+	K:                 4,
+	Eta1:              2,
+	Eta2:              2,
+	Du:                11,
+	Dv:                5,
+
+	KeygenFFI: func(cEntropy *C.uint8_t, cConfig *C.MlKemConfig, cPublicKeys *C.uint8_t, cSecretKeys *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_keygen1024(cEntropy, cConfig, cPublicKeys, cSecretKeys))
+	},
+	EncapsulateFFI: func(cMessage *C.uint8_t, cPublicKeys *C.uint8_t, cConfig *C.MlKemConfig, cCiphertexts *C.uint8_t, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_encapsulate1024(cMessage, cPublicKeys, cConfig, cCiphertexts, cSharedSecrets))
+	},
+	DecapsulateFFI: func(cSecretKeys *C.uint8_t, cCiphertexts *C.uint8_t, cConfig *C.MlKemConfig, cSharedSecrets *C.uint8_t) runtime.EIcicleError {
+		return runtime.EIcicleError(C.icicle_ml_kem_decapsulate1024(cSecretKeys, cCiphertexts, cConfig, cSharedSecrets))
+	},
+}
```

Comment:
Since the available KyberParams are known and aren't customizable, we should change this KyberParams to be an enum of only these 3 possibilities, otherwise the API allows for custom params which isn't supported.

I would also rename to KyberSecLevel or something along those lines as